test/es-module/test-esm-loader-programmatically.mjs

```diff
@@ -0,0 +1,65 @@
+import { spawnPromisified } from '../common/index.mjs';
+import * as fixtures from '../common/fixtures.mjs';
+import assert from 'node:assert';
+import { execPath } from 'node:process';
+import { describe, it } from 'node:test';
+
+// This test ensures that the registerLoader function can register loaders
+// programmatically.
+
+const commonArgs = [
+  '--no-warnings',
+  '--input-type=module',
+  '--loader',
+  'data:text/javascript,',
+]
+
+describe('ESM: register loader programmatically', { concurrency: true }, () => {
+  it('should be able to register programmatically loader', async () => {
+    const { code, signal, stdout, stderr } = await spawnPromisified(execPath, [
+      ...commonArgs,
+      '--eval',
+      ''.concat(`import { registerLoader } from 'node:module';`)
+        .concat(`registerLoader(${JSON.stringify(fixtures.fileURL('es-module-loaders', 'loader-resolve-passthru.mjs'))});`)
+        .concat(`registerLoader(${JSON.stringify(fixtures.fileURL('es-module-loaders', 'loader-load-passthru.mjs'))});`)
+        .concat(`await import('data:text/javascript,console.log(%22Hello from dynamic import%22)');`),
+    ]);
+
+    assert.strictEqual(stderr, '');
+    assert.strictEqual(code, 0);
+    assert.strictEqual(signal, null);
+
+    const lines = stdout.split('\n');
+
+    assert.match(lines[0], /resolve passthru/);
+    assert.match(lines[1], /load passthru/);
+    assert.match(lines[2], /Hello from dynamic import/);
+
+    assert.strictEqual(lines[3], '');
+  });
+
+  it('should be able to register loader programmatically in the same chain of CLI loaders', async () => {
+    const { code, signal, stdout, stderr } = await spawnPromisified(execPath, [
+      ...commonArgs,
+      '--loader',
+      fixtures.fileURL('es-module-loaders', 'loader-resolve-passthru.mjs'),
+      '--eval',
+      ''.concat(`import { registerLoader } from 'node:module';`)
+        .concat(`registerLoader(${JSON.stringify(fixtures.fileURL('es-module-loaders', 'loader-load-passthru.mjs'))});`)
+        .concat(`await import('data:text/javascript,console.log(%22Hello from dynamic import%22)');`),
```

Comment:
nit: this can be DRYed up (pre-computed and cached at the top), which will a make string template nicer too.